store/rootmulti/store.go

```diff
@@ -485,12 +515,14 @@ func (rs *Store) CacheWrapWithTrace(_ io.Writer, _ types.TraceContext) types.Cac
 // CacheMultiStore creates ephemeral branch of the multi-store and returns a CacheMultiStore.
 // It implements the MultiStore interface.
 func (rs *Store) CacheMultiStore() types.CacheMultiStore {
+	rs.listenersMx.Lock()
+	defer rs.listenersMx.Unlock()
```

Comment:
I don't understand why this has to be synchronized. Where is the concurrency?

Reply:
It was synchronized based on review from @peterbourgon (ref: https://github.com/cosmos/cosmos-sdk/pull/14207#discussion_r1084643517). If there is no risk of concurrency then it isn't necessary.

Reply:
There is zero concurrency here. I would rather avoid mutexes if possible, especially in a hot code path like this.

Reply:
I'll revert the change.

Reply:
done